File: pdu/python/geometry_msgs/pdu_cdr_conv_TwistWithCovariance.py
from .pdu_pytype_TwistWithCovariance import TwistWithCovariance
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..geometry_msgs.pdu_cdr_conv_Twist import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_TwistWithCovariance(writer: CdrWriter, src: TwistWithCovariance):
    py_to_cdr_body_Twist(writer, src.twist)
    values = src.covariance
    for i in range(36):
        writer.write_float64(values[i] if i < len(values) else 0.0)


def cdr_body_to_py_TwistWithCovariance(reader: CdrReader, dst: TwistWithCovariance):
    cdr_body_to_py_Twist(reader, dst.twist)
    dst.covariance = [reader.read_float64() for _ in range(36)]
    return dst


def py_to_cdr_TwistWithCovariance(src: TwistWithCovariance) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_TwistWithCovariance(writer, src)
    return writer.bytes()


def cdr_to_py_TwistWithCovariance(cdr_payload) -> TwistWithCovariance:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = TwistWithCovariance()
    return cdr_body_to_py_TwistWithCovariance(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_TwistWithCovariance.js
import { TwistWithCovariance } from './pdu_jstype_TwistWithCovariance.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Twist } from '../geometry_msgs/pdu_jstype_Twist.js';
import { PduTwistConverter } from '../geometry_msgs/pdu_cdr_conv_Twist.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduTwistWithCovarianceConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { TwistWithCovariance } src
     */
    static to_cdr_body(writer, src) {
        PduTwistConverter.to_cdr_body(writer, src.twist);
        {
            const values = src.covariance;
            for (let i = 0; i < 36; i++) {
                writer.write_float64(i < values.length ? values[i] : 0.0);
            }
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { TwistWithCovariance } dst
     * @returns { TwistWithCovariance }
     */
    static cdr_body_to_js(reader, dst) {
        PduTwistConverter.cdr_body_to_js(reader, dst.twist);
        dst.covariance = Array.from({ length: 36 }, () => reader.read_float64());
        return dst;
    }

    /**
     * @param { TwistWithCovariance } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { TwistWithCovariance }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new TwistWithCovariance());
    }
}
